lib/Agrammon/Web/Routes.pm6

```diff
@@ -500,6 +502,26 @@ sub application-routes(Agrammon::Web::Service $ws) {
             }
         }
 
+        post -> LoggedIn $user, 'upload' {
+            request-body -> (:$file, :datasetName($dataset-name), :$comment) {
+                my $file-name = $file.filename;
+                my $content = $file.body-text;
+                my $comment-string = $comment && decode-percents($comment.body-text) || "$file-name uploaded " ~ timestamp;
+                my $lines = $ws.upload-dataset(
+                    $user, decode-percents($dataset-name.body-text),
+                    $content,
+                    $comment-string
+                );
+                content 'application/json', { :$lines };
+                CATCH {
+                    note "$_";
+                    when X::Agrammon::DB::Dataset::UploadFailed {
+                        conflict 'application/json', %( error => .message );
```

Comment:
Hm, is this really a conflict with current state? Seems `bad-request` may be a more appropriate error code.

Reply:
Fixed.